skill-tree/src/main/java/com/RDS/skilltree/viewmodels/SkillRequestViewModel.java

```diff
@@ -0,0 +1,37 @@
+package com.RDS.skilltree.viewmodels;
+
+import com.RDS.skilltree.User.UserModel;
+import com.RDS.skilltree.User.UserSkillsModel;
+import com.RDS.skilltree.models.Endorsement;
+import com.RDS.skilltree.models.Skill;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@NoArgsConstructor
+public class SkillRequestViewModel {
+    private Integer skillId;
+    private String skillName;
+    private String endorseId;
+    private List<MinimalEndorsementViewModel> endorsements;
+
+    public static SkillRequestViewModel toViewModel(
+            UserSkillsModel userSkillsModel, List<Endorsement> endorsements) {
+        SkillRequestViewModel skillRequestViewModel = new SkillRequestViewModel();
```

Comment:
here a new SkillRequestViewModel object is created and then its fields are populated. You can potentially simplify this by using a constructor or setter methods in SkillRequestViewModel that takes the required parameters. *Redundant object creation*

can we use something like below 
public static SkillRequestViewModel toViewModel(UserSkillsModel userSkillsModel, List<Endorsement> endorsements) {
  Skill skill = userSkillsModel.getSkill();
  UserModel user = userSkillsModel.getUser();
  
  return new SkillRequestViewModel(
      skill.getId(),
      skill.getName(),
      user.getId(),
      endorsements.stream()
          .map(MinimalEndorsementViewModel::toViewModel)
          .collect(Collectors.toList()));
}

Reply:
- created a constructor and directly returned a the object after creating it like you suggested:
```java
return new SkillRequestViewModel(
skill.getId(),
skill.getName(),
user.getId(),
endorsements.stream()
.map(MinimalEndorsementViewModel::toViewModel)
.collect(Collectors.toList()));
}
```